File: 121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty())
            return 0;     
        int runMin = prices[0];
        int res = 0;
        for (auto p : prices) {
            res = max(res, p - runMin);
            runMin = min(runMin, p);
        }
        return res;
    }
};

File: 121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1:
            return 0
        currMin = prices[0]
        profit = 0
        for price in prices[1:]:
            profit = max(profit, price - currMin)
            currMin = min(currMin, price)
        return profit
        
